libraries/c_sdk/standard/ble/test/iot_mqtt_ble_system_test.c

```diff
@@ -0,0 +1,1040 @@
+    }
+}
+
+static void setupBleTransportInterface( NetworkContext_t * pContext )
+{
+    bool status;
+
+    /* Static memory that will contain the incoming packet queue used by the transport code. */
+    static uint8_t buffer[ TRANSPORT_BUFFER_SIZE ] = { 0 };
+
+    status = IotBleMqttTransportInit( buffer, TRANSPORT_BUFFER_SIZE, pContext );
+    TEST_ASSERT_TRUE_MESSAGE( status, "Failed to initialize transport interface for BLE" );
+
+    /* Open is a handshake procedure, so we need to wait until it is ready to use. */
+    pContext->pChannel = IotBleDataTransfer_Open( IOT_BLE_DATA_TRANSFER_SERVICE_TYPE_MQTT );
+    TEST_ASSERT_NOT_NULL_MESSAGE( pContext->pChannel, "Failed to create BLE data transfer channel." );
+
+    status = IotSemaphore_Create( &bleChannelSem, 0, 1 );
+    TEST_ASSERT_TRUE( status );
+
+    status = IotBleDataTransfer_SetCallback( pContext->pChannel, bleChannelCallback, NULL );
+    TEST_ASSERT_TRUE( status );
+
+    status = IotSemaphore_TimedWait( &bleChannelSem, TEST_BLE_CONNECTION_TIMEOUT_MS );
+    TEST_ASSERT_TRUE_MESSAGE( status, "Timed out to open BLE data transfer channel" );
+}
+
+static void teardownBleTransportInterface( NetworkContext_t * pContext )
+{
+    IotBleMqttTransportCleanup( pContext );
+    IotBleDataTransfer_Close( pContext->pChannel );
+    IotBleDataTransfer_Reset( pContext->pChannel );
+    IotSemaphore_Destroy( &bleChannelSem );
+}
+
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+static void testSetUp()
+{
+    /* Reset file-scoped global variables. */
+    receivedSubAck = false;
+    receivedUnsubAck = false;
+    receivedPubAck = false;
+    packetTypeForDisconnection = MQTT_PACKET_TYPE_INVALID;
+    memset( &incomingInfo, 0u, sizeof( MQTTPublishInfo_t ) );
+
+    /* setup BLE transport interface. */
+    setupBleTransportInterface( &networkContext );
+
+    /* Establish MQTT session on top of BLE connection. */
+    establishMqttSession( &context, &networkContext, true );
+}
+
+/* Called after each test method. */
+static void testTearDown()
+{
+    /* Terminate MQTT connection. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
+
+    /* Teardown BLE transport interface */
+    teardownBleTransportInterface( &networkContext );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test group for running coreMQTT system tests with a broker
+ * that supports all features of MQTT 3.1.1 specification.
+ */
+TEST_GROUP( coreMQTT_Integration_BLE );
+
+TEST_SETUP( coreMQTT_Integration_BLE )
+{
+    testSetUp();
+}
+
+TEST_TEAR_DOWN( coreMQTT_Integration_BLE )
+{
+    testTearDown();
+}
+
+/* ========================== Test Cases ============================ */
+
+/**
+ * @brief Test group runner for MQTT system tests that run over BLE assisted by a companion device.
+ * These tests are a subset of MQTT integration tests supported over BLE and by AWS IoT.
```

Comment:
How are these tests picked? Are all other remaining tests not supported over BLE?

Reply:
Yes, tests are picked based on all features supported by the companion device SDK.